client/state/data-layer/wpcom/account-recovery/lookup/index.js

```diff
@@ -35,20 +35,20 @@ export const fromApi = data => ( [
 	},
 ] );
 
-export const onSuccess = ( { dispatch }, action, next, data ) => {
-	if ( validate( data ) ) {
-		dispatch( {
-			type: ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE,
-			items: fromApi( data ),
-		} );
-	} else {
+export const storeReceivedOptions = ( { dispatch }, action, next, data ) => {
+	if ( ! validate( data ) ) {
 		throw Error( 'Unexpected response format' );
 	}
+
+	dispatch( {
+		type: ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE,
+		items: fromApi( data ),
```

Comment:
Well, you know I'm not going to let this kind of side effect concealing tip go away 🤖
I've applied the tip in a998687. A bonus is that the `validate` function is now trivial to test, so I've also added a test suite for it.